lib/puppet/type/logical_volume.rb

```diff
@@ -35,6 +35,27 @@
         raise ArgumentError , "#{value} is not a valid logical volume size"
       end
     end
+    def insync?(is)
+        lvm_size_units = { "K" => 1, "M" => 1024, "G" => 1048576, "T" => 1073741824, "P" => 1099511627776, "E" => 1125899906842624 }
+        lvm_size_units_match = lvm_size_units.keys().join('|')
+
+        if is =~ /(\d+\.{0,1}\d{0,2})(#{lvm_size_units_match})/i
```

Comment:
I was looking at this: https://github.com/puppetlabs/puppetlabs-lvm/pull/125/files#diff-92d09665e3a40866889b747edead8e3fL34

I'm less concerned about duplication on this scale, as long as the regexes are consistent.